src/Arrays/CMBuffers.jl

```diff
@@ -98,6 +98,8 @@ function prepare_stage!(buf::CMBuffer)
     if buf.transfer === nothing
         # nothing to do here
     else
+        # This could be `async=true`, since we will launch
+        # a kernel on the same stream afterwards.
         copybuffer!(buf.stage, buf.transfer, async=false)
```

Comment:
@kpamnany did you run into issues with this set to true?

Reply:
No. It should be set to `true`; it works fine and offers a small but measurable speedup.